sound/soc/sof/intel/hda-ipc.c

```diff
@@ -260,11 +261,20 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 		snd_sof_dsp_update_bits_unlocked(sdev, HDA_DSP_BAR,
 						 HDA_DSP_REG_ADSPIC,
 						 HDA_DSP_ADSPIC_IPC, 0);
-		ret = IRQ_WAKE_THREAD;
+		spin_unlock(&sdev->hw_lock);
```

Comment:
careful with the unlock. we could have an IPC and SoundWire interrupt at the same time (different registers used for status), so the two if cases are not mutually exclusive in corner cases

Reply:
Thanks for the reminding, now I separate the handler and thread. So no problem here now.